js/shop.js

```diff
@@ -1,299 +1,312 @@
+   * @param add_course_url
+   * @returns {string} viewer_message_text
+   */
+  function course_selection_get_viewer_banner_text(add_course_url) {
+    var viewer_message_text = '<div class="shop-msg-left"><h1>Students: add ' +
+      'this course to your Crimson Cart in my.harvard' + tooltip_link + '</h1>' +
+      '<p>You will be able to receive notifications, join discussions, watch ' +
+      'lecture videos, and upload assignments. There may be a short delay ' +
+      'after you add the course to your cart. You must click on "enroll" in ' +
+      'my.harvard to be officially enrolled by the Registrar as a Student in ' +
+      'this course.</p></div><div class="shop-btn-right">' +
+      '<a class="btn btn-small btn-primary" href="' + add_course_url + '">' +
+      'Add to Crimson Cart</a></div>';
+    return viewer_message_text;
+  }
 
+  /**
+   * Get the banner text for teachers
+   * @returns {string} course_selection_is_active_message
+   */
+  function course_selection_get_teacher_banner_text() {
+    var course_selection_is_active_message = '<h1>Your current class list may include Prospective Enrollees. ' + tooltip_link +
+      '</h1><p>All Harvard ID holders can view this course site during course selection period. Students ' +
+      'can choose to add this course to their Crimson Cart to participate in discussions, upload assignments, watch ' +
+      'lecture videos, and receive notifications for this course before they are officially enrolled through the ' +
+      'Registrar. All Student contributions will be visible to other students who are also participating in this ' +
+      'course. At the end of the course selection period, Prospective Enrollees who have not officially enrolled as ' +
+      'Students or Guests in the course through the Registrar’s office will be removed from the class list.</p>';
+    return course_selection_is_active_message;
+  }
 
-/**
- * allowed_terms is a white list of Canvas enrollment_term_ids where shopping is allowed
- * for a specific school
- * NOTE - the term ids in allowed_terms must be strings, not ints
- * @type {string[]}
- */
+  /**
+   * Common course selection code
+   */
 
-/**
- * terms for dev
- * var allowed_terms = ['3', '4'];
- */
+  var current_user_id = ENV['current_user_id'];
+  var user_url = '/api/v1/users/' + current_user_id + '/profile';
+  var course_id = get_course_id();
+  var course_url = '/api/v1/courses/' + course_id;
+  var login_url = window.location.origin+"/login";
 
-/**
- * updated allowed terms with correct terms before deploying to prod
- */
-var allowed_terms = ['39'];
+  /**
+   * Tool tip text and html link
+   * @type {string}
+   */
+  var course_selection_help_doc_url = 'https://wiki.harvard.edu/confluence/pages/viewpage.action?pageId=168134774';
+  var data_tooltip = 'More info about access during course selection period';
+  var tooltip_link = '<a data-tooltip title="' + data_tooltip + '" target="_blank" href="' +
```

Comment:
Just a stylistic comment, I'm used to seeing things defined before they are used. so seeing this below where it's used is a bit confusing. Same it true for all the vars used this way add_course_url, remove_course_url etc...